### PR TITLE
Cleaned up previous pull request; Prefix, comment additions, typos

### PR DESCRIPTION
**Pull request by daved**
The commits are now cleaner and not in the master branch.  Please let me know if I should keep to my own fork instead.  Thanks!

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func init() {
     }
 }
 
+// MarshalJSON is generated.
```
Comment (campoy) [inline template.go]: The convention for these comments is:

```go
// {{$typename}} is a json.Marshaler.
```

File: template.go [hunk 2]
```diff
@@ -53,6 +54,7 @@ func (r {{$typename}}) MarshalJSON() ([]byte, error) {
     return json.Marshal(s)
 }
 
+// UnmarshalJSON is generated.
```
Comment (campoy) [inline template.go]: The convention for these comments is:

```go
// {{$typename}} is a json.Unmarshaler.
```

**Comment by codingmusica** [comment]
golint complains - ` comment on exported method AppID.MarshalJSON should be of the form "MarshalJSON ..."`

I apologize if I'm misunderstanding something.

**Comment by campoy** [comment]
Ah, indeed. Maybe:

```go
// MarshalJSON is generated so {{$typename}} satisfies json.Marshaler.
```

**Comment by codingmusica** [comment]
Corrected.

**Comment by shurcooL** [comment]
Any updates here?

**Comment by campoy** [comment]
Sorry I missed this.

**Pull request merged**